WordPress/openverse-frontend
# Add utilities for consistent focus rings

## Fixes




Related to the comment https://github.com/WordPress/openverse-frontend/issues/2049#issuecomment-1358044819 by @panchovm

## Description




This PR
- defines new Tailwind utilies to use for focus rings
- adds some stories that use these utilities
- adds visual regression tests for the 'Focus' story
- updates a few usages of each of the three utilities as an example
  - updates the `VCheckbox` component to match the WordPress component library
  - updates the search result image cell so that the white outline masks content instead of appearing outside
  - updates the search result audio cell so that the white outline masks content instead of appearing outside

This PR does not
- update every component to use these utilities
- remove other implementations of focus rings

These new utilities do not include any transitions on the shadows. This is intentional as the focus movement reacting to the <kbd>Tab</kbd> must be as fast as possible. If needed in some specific cases, it can be restored with the `.transition-shadow` utility.

## Testing Instructions



0. Check out the PR and run Storybook.
1. See the stories under Meta/Focus.
2. Add some rounded styles like `rounded-sm`, `rounded` and `rounded-full` to see how the ring behaves.
3. Read the docs for Meta/Focus.

---

0. Check out the PR and run the dev server.
1. Use the keyboard navigation to focus on the following three components:
	i. <s>Filters button in the new header (slim filled style)</s>
	ii. Checkboxes in the filters sidebar (slim transparent style with a slight mod)
	iii. Image cell in the search results (bold filled style)
    iv. Audio cell in the search results (bold filled style)
    v. Audio row in the search results (slim transparent style)

## Screenshots

![image](https://user-images.githubusercontent.com/16580576/209108622-ece84eff-9cb6-49c8-8098-6c29cc6a4d93.png)

With additional `rounded-full` style using the 'Controls' panel.

## Checklist



      `Update index.md`).
      a parent feature branch.
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin



<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
